base/operators.jl

```diff
@@ -964,6 +964,11 @@ reverse{A,B}(p::Pair{A,B}) = Pair{B,A}(p.second, p.first)
 endof(p::Pair) = 2
 length(p::Pair) = 2
 
+convert{A,B}(::Type{Pair{A,B}}, x::Pair{A,B}) = x
+function convert{A,B,C,D}(::Type{Pair{A,B}}, x::Pair{C,D})
```

Comment:
It should be sufficient to declare `function convert{A,B}(::Type{Pair{A,B}}, x::Pair)` since you don't use `C` or `D`